### PR TITLE
Refactor: remove use of AppConfig in AppRevision

### PR DESCRIPTION
**Pull request by wonderflow**
1. This PR refactors the AppRevision format, AppConfig and Components was deprecated and will not be used in the code logic.
2. During the new design and workflow feature, the config and secret passing mechanism in Definition will be removed.

### CODE REVIEW

**Comment by leejanee** [comment]
when use `input/output` in application components, the incomplete parameter will cause ` af.GenerateComponentManifest()` failed.

File: pkg/appfile/parser_test.go [hunk 1]
```diff
@@ -22,21 +22,21 @@ import (
 	"reflect"
 	"strings"
 
+	"github.com/google/go-cmp/cmp"
+
```
Comment (hongchaodeng) [inline pkg/appfile/parser_test.go:26]: format import...

File: apis/core.oam.dev/v1beta1/applicationrevision_types.go [hunk 2]
```diff
@@ -43,14 +43,25 @@ type ApplicationRevisionSpec struct {
...
+	// +deprecated
 	Components []common.RawComponent `json:"components,omitempty"`
 
 	// ApplicationConfiguration records the rendered applicationConfiguration from Application,
 	// it will contains the whole K8s CR of trait and the reference component in it.
 	// +kubebuilder:validation:EmbeddedResource
 	// +kubebuilder:pruning:PreserveUnknownFields
-	ApplicationConfiguration runtime.RawExtension `json:"applicationConfiguration"`
+	// +deprecated
+	ApplicationConfiguration runtime.RawExtension `json:"applicationConfiguration,omitempty"`
```
Comment (hongchaodeng) [inline apis/core.oam.dev/v1beta1/applicationrevision_types.go:64]: I can't think of anyone using it from AppRevision.
Let's just remove this field.

File: apis/types/types.go [hunk 3]
```diff
@@ -71,3 +73,12 @@ const (
 	// TypePlugin defines one category used in Kubectl Plugin
 	TypePlugin = "Plugin Command"
 )
+
+// DefaultFilterAnnots are annotations won't pass to workload or trait
```
Comment (hongchaodeng) [inline apis/types/types.go:77]:
```suggestion
// DefaultFilterAnnots are annotations that won't pass to workload or trait
```

**Review by hongchaodeng — APPROVED** [review]
This is awesome!

LGTM in general

File: apis/core.oam.dev/v1beta1/applicationrevision_types.go (hunk 2 above)
Comment (wonderflow) [reply apis/core.oam.dev/v1beta1/applicationrevision_types.go:64]: Let's remove it next time. It's required value before..

**Comment by wonderflow** [comment]
closing as it's contained in #2236

**Pull request closed without merging**